metadata-service/auth-ranger-impl/src/main/java/com/datahub/authorization/ranger/RangerAuthorizer.java

```diff
@@ -0,0 +1,89 @@
+package com.datahub.authorization.ranger;
+
+import com.datahub.authorization.AuthorizationRequest;
+import com.datahub.authorization.AuthorizationResult;
+import com.datahub.authorization.AuthorizedActors;
+import com.datahub.authorization.Authorizer;
+import com.datahub.authorization.AuthorizerContext;
+import com.datahub.authorization.ResourceSpec;
+import com.linkedin.common.urn.Urn;
+import com.linkedin.common.urn.UrnUtils;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import javax.annotation.Nonnull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.ranger.plugin.policyengine.RangerAccessRequest;
+import org.apache.ranger.plugin.policyengine.RangerAccessRequestImpl;
+import org.apache.ranger.plugin.policyengine.RangerAccessResourceImpl;
+import org.apache.ranger.plugin.policyengine.RangerAccessResult;
+
+
+@Slf4j
+public class RangerAuthorizer implements Authorizer {
+
+  private AuthorizerConfig authorizerConfig;
+  private DataHubRangerClient dataHubRangerClient;
+
+  public RangerAuthorizer() {
+  }
+
+  @Override
+  public void init(@Nonnull Map<String, Object> authorizerConfigMap, @Nonnull final AuthorizerContext ctx) {
+    this.authorizerConfig = AuthorizerConfig.builder()
+        .username((String) authorizerConfigMap.get(AuthorizerConfig.CONFIG_USERNAME))
+        .password((String) authorizerConfigMap.get(AuthorizerConfig.CONFIG_PASSWORD))
+        .build();
+
+    this.dataHubRangerClient = this.newDataHubRangerClient();
+    this.dataHubRangerClient.init();
+  }
+
+  public DataHubRangerClient newDataHubRangerClient() {
+    return new DataHubRangerClient(this.authorizerConfig);
+  }
+
+  @Override
+  public AuthorizationResult authorize(AuthorizationRequest request) {
+
+    String userIdentifier = UrnUtils.getUrn(request.getActorUrn()).getId();
+
+    Set<String> roles = this.dataHubRangerClient.getUserRoles(userIdentifier);
+    Set<String> groups = this.dataHubRangerClient.getUserGroups(userIdentifier);
```

Comment:
It is EXTREMELY important that the way we do this join from DataHub User to Privacera user is well understood.